theme/theme.go

```diff
@@ -327,6 +332,8 @@ func (t *builtinTheme) Size(s fyne.ThemeSizeName) float32 {
 		return 11
 	case SizeNameInputBorder:
 		return 2
+	case SizeNameTabWidth:
```

Comment:
I'm not sure about this name, or maybe even if this is the right place?
All the other sizes are in "dp" so 4 is really small (=Padding).
And then I started thinking, should it maybe be a user preference instead of a theme preference?
I'm not convinced by either right now, should we set '4' as a constant for "TabSpaceWidth" or something in the painter package for now?

Reply:
I was not sure about it when I did it. First, I wanted it in the TextStyle, but that does not make it down to the level of the font drawing. I then chose the Theme as you seem to want to control all the aspects of the appearance via that, and handily it is globally available.
The argument of tab width and if a tab is a user or developer setting is another can of worms. i.e if I was writing a GO editor, as a developer I would want to set the tab width to 8. But if I was writing a node.js app I would want it set to 2. If I was writing a text editor I would want it as a user preference. I might even want to change it depending on what I have open and even have multiple different sizes. 

I don't think it should be a constant and should at least be changeable by the developer. I think you should decide where you want it as you have a better understanding of Fyne than me.

Reply:
I agree with your summary. The API would probably be needed in the Entry/Label so it can be exposed to the developer.
To do this it may be required to set it on Text - I will leave it to see what you think.

Reply:
I had forgotton about Fyne.MeasureText() It is what made me chose the theme before. As it has nothing to do with Label or Entry setting up tabWidth in textProvider wont work.
Options:

- Stick with theme.
- Use user preferences but hard code a default of 4. I don't see this gives any advantages vs the theme.
- Modify the fyne.MeasureText() call to also require tabWidth. Breaks the API
- Add tabWidth to textStyle. This would be my choice.

Reply:
1 vs 2 - it depends if we think this is user preference or developer preference. Users cannot customise theme beyone primary colour and light/dark and even that is optional for themes.
Following 3 seems like a breaking change that is not needed.
4 could be a smart move, seems consistent with the use of that to configure text in other ways.

Reply:
TabWidth is now in the TextStyle